src/storage/src/hummock/state_store.rs

```diff
@@ -116,6 +118,18 @@ impl HummockStorage {
                 .await?,
             ));
         }
+        while let Some(sync_uncommitted_data) = sync_uncommitted_datas.pop() {
```

Comment:
This looks weird. Can we use a reversed into iterator instead of popping items from the while loop?